File: Longest Consecutive Sequence.py
#Time Complexity: O(N)
#Space Complexity: O(N)

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        maxSequence = 0

        for num in nums:
            if (num - 1) not in numSet:
                numberSequence = 0
                while (num + numberSequence) in numSet:
                    numberSequence += 1
                maxSequence = max(maxSequence, numberSequence)
        return maxSequence

File: Longest Consecutive Sequence.js
//Time O(n)
//Space O(n)

/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const set = new Set(nums);
    let maxSequence = 0;

    for (let num of set) {
        if (!set.has(num - 1)) {
            let numberSequence = 0;
            while (set.has(num + numberSequence)) {
                numberSequence += 1;
            }
            maxSequence = Math.max(maxSequence, numberSequence);
        }
    }
    return maxSequence;
};
